src/components/MyAccount/NewSettings/PasswordForm.tsx

```diff
@@ -0,0 +1,262 @@
+  Banner,
+  Flex,
+  SkeletonLoader,
+  Text,
+  TextInput,
+} from "@nypl/design-system-react-components"
+import SettingsLabel from "./SettingsLabel"
+import SaveCancelButtons from "./SaveCancelButtons"
+import type { Patron } from "../../../types/myAccountTypes"
+import { BASE_URL } from "../../../config/constants"
+import EditButton from "./EditButton"
+
+interface PasswordFormProps {
+  patronData: Patron
+  settingsState
+}
+
+const PasswordForm = ({ patronData, settingsState }: PasswordFormProps) => {
+  const { getMostUpdatedSierraAccountData } = useContext(PatronDataContext)
+  const [isLoading, setIsLoading] = useState(false)
+  const [isEditing, setIsEditing] = useState(false)
+  const [formData, setFormData] = useState({
+    oldPassword: "",
+    newPassword: "",
+    confirmPassword: "",
+    passwordsMatch: true,
+  })
+  const { setIsSuccess, setIsFailure, isOtherEditing, setIsOtherEditing } =
+    settingsState
+
+  const cancelEditing = () => {
+    setIsEditing(false)
+    setIsOtherEditing(false)
+  }
+
+  const validateForm =
+    formData.oldPassword !== "" &&
+    formData.newPassword !== "" &&
+    formData.confirmPassword !== "" &&
+    formData.passwordsMatch
+
+  const handleInputChange = (e) => {
+    const { name, value } = e.target
+    let updatedFormData = { ...formData }
+
+    if (name === "confirmPassword") {
+      updatedFormData = {
+        ...updatedFormData,
+        confirmPassword: value,
+        passwordsMatch: updatedFormData.newPassword === value,
+      }
+    } else if (name === "newPassword") {
+      updatedFormData = {
+        ...updatedFormData,
+        newPassword: value,
+        passwordsMatch: updatedFormData.confirmPassword === value,
+      }
+    } else {
+      updatedFormData = {
+        ...updatedFormData,
+        [name]: value,
+      }
+    }
+    setFormData(updatedFormData)
+  }
+
+  const submitForm = async () => {
+    setIsLoading(true)
+    setIsEditing(false)
+    setIsSuccess(false)
+    setIsFailure(false)
+    try {
+      const response = await fetch(
+        `${BASE_URL}/api/account/update-pin/${patronData.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            oldPin: formData.oldPassword,
+            newPin: formData.newPassword,
+            barcode: patronData.barcode,
+          }),
+        }
+      )
+
+      if (response.status === 200) {
+        setIsSuccess(true)
+        await getMostUpdatedSierraAccountData()
+      } else {
+        setIsFailure(true)
+      }
+    } catch (error) {
+      console.error("Error submitting", error)
+    } finally {
+      setIsLoading(false)
+      setIsOtherEditing(false)
+    }
+  }
+
+  return (
+    <>
+      {isLoading ? (
+        <SkeletonLoader contentSize={2} showImage={false} headingSize={0} />
+      ) : isEditing ? (
+        <>
+          <Flex flexDir={{ base: "column", lg: "row" }}>
```

Comment:
Noting that L111-190 is extremely repetitive but also resistant to being pulled out as a component... the `onChange`s are sensitive I guess